src/components/states/emptyState/emptyState.test.tsx

```diff
@@ -0,0 +1,105 @@
+import { render, screen } from '@testing-library/react';
+import { IconType } from '../../icon';
+import { EmptyState } from './emptyState';
+import type { IEmptyStateProps } from './emptyState.api';
+
+describe('<EmptyState /> component', () => {
+    const createTestComponent = (props?: Partial<IEmptyStateProps>) => {
+        const commonProps = {
+            heading: 'test-heading',
+        };
+
+        if (props?.humanIllustration) {
+            return <EmptyState humanIllustration={props.humanIllustration} {...commonProps} {...props} />;
+        } else {
+            const { humanIllustration, objectIllustration = { object: 'ACTION' }, ...otherProps } = props ?? {};
+
+            return <EmptyState objectIllustration={objectIllustration} {...commonProps} {...otherProps} />;
+        }
+    };
+
+    it('renders the EmptyState component stacked with full props and object illustration', () => {
+        const objectIllustration = { object: 'LIGHTBULB' } as const;
+        const primaryButton = {
+            label: 'Label',
+            iconLeft: IconType.ADD,
+            iconRight: IconType.CHEVRON_RIGHT,
+        } as const;
+        const secondaryButton = {
+            label: 'Label',
+            iconLeft: IconType.ADD,
+            iconRight: IconType.CHEVRON_RIGHT,
+        } as const;
```

Comment:
If we're not testing the `iconLeft`,`iconRight` properties of the button (and we shouldn't as this logic is specific to the button component), we don't need to specify them. Same for the tests below
```suggestion
const primaryButton = { label: 'Label' };
const secondaryButton = { label: 'label' };
```